### PR TITLE
doc/install: refine index.rst

### PR DESCRIPTION
**Pull request by zdover23**
Refine English sentences in doc/install/index.rst. Remove adverbial phrases of time that refer to Nautilus-era features as "new", since that was four years ago.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/install/index.rst [hunk 1]
```diff
@@ -4,34 +4,33 @@
 Installing Ceph
 ===============
 
-There are several different ways to install Ceph.  Choose the
-method that best suits your needs.
+There are several different ways to install Ceph.  Choose the method that best
```
Comment (anthonyeleven) [inline doc/install/index.rst:7]: I would write "multiple"

File: doc/install/index.rst [hunk 2]
```diff
@@ -4,34 +4,33 @@
...
-  fully supports the new CLI and dashboard features to manage
-  cluster deployment.
-* cephadm requires container support (podman or docker) and
+cluster that uses containers and systemd and is tightly integrated with the CLI
+and dashboard GUI.
+
+* cephadm supports only Octopus and newer releases.
+* cephadm is fully integrated with the orchestration API and fully supports the
+  CLI and dashboard features that are used to manage cluster deployment.
+* cephadm requires container support (in the form of podman or docker) and
```
Comment (anthonyeleven) [inline doc/install/index.rst:20]: Podman or Docker?

File: doc/install/index.rst [hunk 3]
```diff
@@ -40,16 +39,20 @@ Other methods
 Ceph clusters using Ansible.
 
 * ceph-ansible is widely deployed.
-* ceph-ansible is not integrated with the new orchestrator APIs,
-  introduced in Nautilus and Octopus, which means that newer
-  management features and dashboard integration are not available.
+* ceph-ansible is not integrated with the orchestrator APIs introduced in
+  Nautilus and Octopus, which means that the management features and dashboard
+  integration they introduced are not available in Ceph clusters deployed by
```
Comment (anthonyeleven) [inline doc/install/index.rst:44]: which means that these API management features and dashboard integration are not available in Ceph clusters

To me the antecedent of "they" is ambiguous here.

**Review by anthonyeleven — APPROVED** [review]
Non-blocking suggestions

File: doc/install/index.rst (hunk 2 above)
Comment (zdover23) [reply doc/install/index.rst:20]: Yeah, capital letters are preferable.

File: doc/install/index.rst (hunk 3 above)
Comment (zdover23) [reply doc/install/index.rst:44]: Fair enough. I'll recast. it.

File: doc/install/index.rst (hunk 1 above)
Comment (zdover23) [reply doc/install/index.rst:7]: It is pithier. I agree.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/50435 - Quincy backport
https://github.com/ceph/ceph/pull/50436 - Pacific backport

Reef requires no backport because these changes went into that branch by another means too irritating to express here.

**Pull request merged**